core/modules/wmgr/python/dbMgr.py

```diff
@@ -604,8 +639,91 @@ def locadData(dbName, tblName, chunkId=None):
     _validateDbName(dbName)
     _validateTableName(tblName)
 
-    _log.error('method not implemented yet')
-    raise ExceptionResponse(501, "NotImplemented", "Data loading operations are not implemented yet")
+    # check that table exists
+    dbConn = Config.instance().dbConn()
+    if not dbConn.tableExists(tblName, dbName):
+        raise ExceptionResponse(404, "TableMissing", "Table %s.%s does not exist" % (dbName, tblName))
+
+    # determine chunk table name (if loading to chunk)
+    if chunkId is not None:
+        if request.path.endswith('/overlap'):
+            tblName = tblName + 'FullOverlap_' + str(chunkId)
+        else:
+            tblName = tblName + '_' + str(chunkId)
+        if not dbConn.tableExists(tblName, dbName):
+            raise ExceptionResponse(404, "ChunkTableMissing",
+                                    "Chunk table %s.%s does not exist" % (dbName, tblName))
+
+    # optional load-options
+    options = dict(delimiter='\t', enclose='"', escape='\\', terminate='\n',
+                   compressed=None, local="0")
+    formOptions = request.form.get('load-options')
+    _log.debug('formOptions: %s', formOptions)
+    if formOptions:
+        formOptions = url_decode(formOptions)
+
+        # check that there are no non-recognized options
+        for key, value in formOptions.items():
+            if key not in options:
+                raise ExceptionResponse(400, "IllegalOption",
+                                        "unrecognized option %s in load-options" % key)
+            options[key] = value
+        _log.debug('options: %s', options)
+
+    # get table data
+    data = request.files.get('table-data')
+    if data is None:
+        raise ExceptionResponse(400, "MissingFileData", "table-data part is missing from request")
+    _log.debug('data: name=%s filename=%s', data.name, data.filename)
+
+    # named pipe to send data to mysql, make it in a temporary dir
+    tmpDir = tempfile.mkdtemp()
+    fifoName = os.path.join(tmpDir, 'tabledata.dat')
+    os.mkfifo(fifoName, 0600)
+
+    # do we need to uncompress?
+    compressed = _getArgFlag(options, 'compressed', None)
+    if compressed is None:
+        compressed = data.filename.endswith('.gz')
```

Comment:
`data.filename` is the name that client passes in a request, workzeug probably stores data in file with different name (or keeps it in memory).